tests/test_core.py

```diff
@@ -4741,6 +4741,38 @@ def test_unicode_js_library(self):
     self.emcc_args += ['--js-library', path_from_root('tests', 'unicode_library.js')]
     self.do_run(open(os.path.join(self.get_dir(), 'main.cpp'), 'r').read(), u'Unicode snowman \u2603 says hello!')
 
+  def test_js_lib_dep_memset(self):
+    open('lib.js', 'w').write(r'''
+mergeInto(LibraryManager.library, {
+  depper__deps: ['memset'],
```

Comment:
Btw, this interacts with what was observed in #2908. Isn't it still the case that JS code cannot add dependencies to  C code unless these dependencies are specified in `src/deps_info.json`? (or was that fixed at some point?)

If that's correct, then, say for clarity, if we replace `memset` by any other function that user has implemented in C side `myFunctionOnCSide`, then it's incorrect to specify `depper__deps: ['myFunctionOnCSide']` since that won't currently do what user intended due to #2908, but user would have to specify the dependency `depper-> myFunctionOnCSide ` in the system file `src/deps_info.json` for it to work?